lib/knex-builder/internal/parse-connection.js

```diff
@@ -32,24 +32,45 @@ module.exports = function parseConnectionString(str) {
 
   const isPG = ['postgresql', 'postgres'].includes(protocol);
 
-  return {
+  const result = {
     client: protocol,
-    connection: isPG ? parsePG(str) : connectionObject(parsed),
+    connection: isPG ? parsePG(str) : connectionObject(protocol, parsed),
   };
+  if (
```

Comment:
This is mixing of parameters for db and parameters for knex, its make harder to understand where options actually will be applied.  
Also `useNullAsDefault` is option to configurate via code, not via connection string config, because code could rely on it. (I'm implying that connection string goes from external config, which is common case. In case of rest options such as `useNullAsDefault`, they should be set from code, because depends on it)